### ISSUE TITLE
Async methods can't be awaited using built in c# await operator

### ISSUE THREAD

**Issue by maximkornilov** [Microsoft/AppCenter-SDK-Unity#250]
Base on the [Asynchronous APIs in the Unity SDK](https://docs.microsoft.com/en-us/appcenter/sdk/unity-async) documentation, it is expected that the App Center SDK's async methods support built-in await operator in case of a codebase has access to .NET 4.6 or above.

But currently, the following code can't be compiled: 
```
private async void Start()
{
    bool isEnabled = await AppCenter.IsEnabledAsync();
}
```

The reported error is: 
```
error CS1061: 'AppCenterTask<bool>' does not contain a definition for 'GetAwaiter' and no accessible extension method 'GetAwaiter' accepting a first argument of type 'AppCenterTask<bool>' could be found (are you missing a using directive or an assembly reference?)
```
### Steps to reproduce
* Open App Center Demo project
* Update player settings configuration to use the following values:
 **Scripting Runtime Version** -> .Net 4x Equivalent
 **Api Compatability Level** -> .Net Standard 2.0 
* Try to consume SDK's async methods using built-in c# features as in code example above.

**Actual**: failed to compile with error described above
**Expected**: async method should be awaited or documentation should be updated to remove misleading information.

**Comment by MatkovIvan** [comment]
Hi Maxim,

Yes, it's expected to work.
As I see in the implementation, this method is applied only for .NET 4.6, non-editor environment:
https://github.com/Microsoft/AppCenter-SDK-Unity/blob/9fe8b5389d25175e8714ac20f0c5cc193763753f/Assets/AppCenter/Plugins/AppCenterSDK/Core/Shared/AppCenterTaskAwaiter.cs#L5

I wonder if you're trying to compile it for an editor?

**Comment by maximkornilov** [comment]
@MatkovIvan , I don't compile it for an editor. 

Basically, the condition you use to verify for supported API is incorrect. At least, it looks so to me. As far as I remember, NET_4_6 is not defined by Unity in case of the `Api Compatability Level` is set to `Net Standard 2.0`. 

One of the possible ways to check for this is: 
```
#define HAS_ASYNC_AWAIT_SUPPORT = CSHARP_7_OR_LATER || (UNITY_2018_3_OR_NEWER && (NET_STANDARD_2_0 || NET_4_6))
#if HAS_ASYNC_AWAIT_SUPPORT && !UNITY_EDITOR
```

This one handles all possible configurations which support built-in async await support in Unity.

**Comment by MatkovIvan** [comment]
Yeah, I agree that this condition doesn't look correct. It was added when Unity didn't support `.Net Standard 2.0`. Thank you for the report and fix suggestion, we'll change this one (but we are always happy to see the community PRs 😉)

**Comment by annakocheshkova** [comment]
Hi @maximkornilov! The bugfix for this has been released with [Unity SDK 1.4.1](https://github.com/Microsoft/AppCenter-SDK-Unity/releases/tag/1.4.1). I'm closing this issue.
